File: csharp/isogram/Isogram.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Isogram
{
  public static bool IsIsogram(string word)
  {
    var charsFound = new List<char>();
    foreach (char c in word)
    {
      if (char.IsLetter(c))
      {
        if (charsFound.Contains(char.ToLower(c)))
        {
          return false;
        }
        charsFound.Add(char.ToLower(c));
      }
    }
    return true;
  }
}

File: go/isogram/isogram.go
package isogram

import (
	"unicode"
)

//IsIsogram returns true if a word is an isogram
func IsIsogram(word string) bool {
	seen := make(map[rune]bool)

	for _, v := range word {
		if string(v) == " " || string(v) == "-" {
			continue
		}
		v = unicode.ToLower(v)
		if seen[v] {
			return false
		}
		seen[v] = true
	}
	return true
}
